contracts/strategies/modules/voting/IVotingModule.sol

```diff
@@ -0,0 +1,19 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.19;
+
+import {IStrategy} from "../../IStrategy.sol";
+
+interface IVotingModule is IStrategy {
+    function initializeVotingModule(bytes _data) external;
+    // initialize the voting itself, which could contain things like an ERC20 used to determine vote count
+
+    function vote(bytes memory _data, address _sender) external payable;
+    // called via allo.sol by users to allocate votes to a recipient
+    // this will update some data in this contract to store votes, etc.
+
+    function getResults(address[] memory _recipientId, bytes memory _data)
```

Comment:
Still finding it unclear on the benefits of having a voting and allocation module. Could we draw out examples on why this would. be needed  ? 
as opposed to having them merged together